omniledger/service/contracts.go

```diff
@@ -247,22 +238,31 @@ func (s *Service) spawnContractConfig(cdb CollectionView, inst Instruction, coin
 		return
 	}
 
+	id := d.GetBaseID()
 	return []StateChange{
-		NewStateChange(Create, GenesisReferenceID, ContractConfigID, inst.InstanceID.DarcID),
-		NewStateChange(Create, inst.InstanceID, ContractDarcID, darcBuf),
-		NewStateChange(Create,
-			InstanceID{
-				DarcID: inst.InstanceID.DarcID,
-				SubID:  oneSubID,
-			}, ContractConfigID, configBuf),
+		NewStateChange(Create, GenesisReferenceID, ContractConfigID, id, id),
```

Comment:
It looks strange (looked strange before, too, so this might be a new issue and not to be solved here) to have two `ContractConfigID` here. In the old system, I can see why we would want to have the `GenesisReferenceID` point to the `ContractConfigID`, but now I think we can do simpler:

```
GenesisReferenceID, ContractConfigID, configBuf, id
InstanceIDFromSlice(id), ContractDarcID, darcBuf, id
```

and then `s/GenesisReferenceID/GenesisConfigID/g`.

This would also remove the `DeriveConfigID`, which should've been `DeriveID(ContractConfigID)`.